source/common/stats/thread_local_store.cc

```diff
@@ -360,13 +361,13 @@ void ParentHistogramImpl::recordValue(uint64_t value) {
 }
 
 bool ParentHistogramImpl::used() const {
-  Thread::LockGuard lock(merge_lock_);
-  return usedLockHeld();
+  // Consider ParentHistogram used only if has ever been merged.
+  return merged_;
 }
 
 void ParentHistogramImpl::merge() {
   Thread::ReleasableLockGuard lock(merge_lock_);
-  if (usedLockHeld()) {
+  if (merged_ || usedLockHeld()) {
```

Comment:
This would ensure that we check merged first, if it is merged, do not bother to look for tls histograms. TLS histogram iteration would only be needed till it is merged first time.